pytorch_lightning/trainer/trainer.py

```diff
@@ -2364,6 +2369,151 @@ def terminate_on_nan(self, val: bool) -> None:
         )
         self._terminate_on_nan = val  # : 212
 
+    def reset_train_dataloader(self, model: Optional["pl.LightningModule"] = None) -> None:
```

Comment:
how about moving the reload logic to data_connector and `trainer.reset_train_dataloader` just calls `trainer._data_connector._reset_train_dataloader`? not sure if it's a good idea, but just trying to avoid all the logic being kept inside the single Trainer module.

Reply:
I don't feel strongly either way. I probably have a slight preference for keeping it as it is now, since I feel like having a function that just calls the private function adds a bit of unnecessary abstraction/complexity. If others feel strongly we can change it

Reply:
I'd say it's fine as it is. I'd only do that if it aids in testing, but we don't have unit tests for each of these separate methods.

Reply:
It is fine because the trainer owns the dataloaders, so resetting is her responsibility. But this won't be the case in the future, pretty sure. Let's just please not mix properties and methods.

Reply:
> n any case, let's just please not mix properties and methods like this.


@daniellepintz I believe my comment was missed here. Could you send a follow up PR? I believe reviewers did not catch this properly

Reply:
@awaelchli sorry I am confused by that line. Could you clarify which property you are talking about?

Reply:
I mean the disorganization that this has caused. Previously, methods and properties were organized, but now new methods have been added and now it's mixed up:

method
method
property
property
new method
new method

It should be

method
method
new method
new method
property
property

given that Trainer wants to have all properties at the bottom.

Reply:
Ah I see good point. Will send a follow up PR